### DISCUSSION TITLE
Is there a way to list all resources matching certain labels?

### DISCUSSION THREAD

**Discussion by gbhat618** [fabric8io/kubernetes-client#6614] [Q&A] [answered]
So far I am only finding examples and apis for specifically listing, pods, services, but I wanted to list all resources that matched certain label.
Essentially the output of,
```
kubectl get all -l label1=value1 --all-namespaces
```

**Comment by manusa** [comment, marked as answer]
Same as #1154 

This feature is not available.

What kubectl does under the hood is check each available resource endpoint with the given label constraint by performing `n` HTTP requests.

You can more or less build the equivalent yourself by doing `Client.getAPIVersions` and then for each a `Client.genericKubernetesResources` and provide the label filter.
